### PR TITLE
Browse mode: Add an H2 inside the sidebar navigation screen title

### LINKED ISSUE #47217: Browse Mode: Use heading elements for the "visual headings"
### Description

In the navigation sidebar in the browse mode there are a few texts that look like headings: They are using a larger text size and are positioned above and in close proximity to other content; the words design, templates and template parts.

These need to be actual heading elements to allow navigating via headings.
For a more detailed explanation of why this is needed please see #42373 and #42495 which were opened before the introduction of the browse mode.

This is in addition to the missing H1 that is already reported.

### Step-by-step reproduction instructions

In the WordPress admin, select Appearance > Editor. 
Once in the Site Editor, confirm that there are no headings on the page.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by carolinan**
## What?

Places the words design, templates and template parts inside `h2` elements in the navigation sidebar in the Site Editor.
Closes https://github.com/WordPress/gutenberg/issues/47217

## Why?

Text that visually looks like headings should use HTML heading elements.

By using heading elements for the visual heading texts instead of divs, users can use the headings to navigate to the content; the list of buttons that represents the templates and template parts.

## How?

Updates the sidebar navigation screen title div:
Adds a new condition for displaying either the heading text or the inner elements that include the
the back button, the template/template parts heading and the add new button.

## Testing Instructions
1.  Open the Site Editor.
2. Confirm that there is no visual change to the sidebar.
3. View the source and confirm that the word Design is placed inside an `H2`.
4. Select either templates or template parts.
5. On the navigation screen that you just opened, confirm that there is no visual change and that the text templates or template parts are inside an `H2`.

### Testing Instructions for Screen reader

1. Open the Site Editor
2. Use the relevant command for your screen reader to navigate to the next heading.
3. This should find a H2 heading with the text Design. 
4. Next navigate forwards to the button with the text Templates. Activate the button.
5. Use the relevant command for your screen reader to navigate to the next heading.
6. This should find a H2 heading with the text Templates. With this, the test is completed.

Note that this document does not have an H1. This will be implemented in a separate PR.

### CODE REVIEW

File: packages/edit-site/src/components/sidebar-navigation-screen/index.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ export default function SidebarNavigationScreen( {
 	parentTitle,
 	title,
 	content,
+	heading,
```
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar-navigation-screen/index.js]: Why did you add a new prop instead of just using the existing "title" prop?

Comment (carolinan) [reply packages/edit-site/src/components/sidebar-navigation-screen/index.js]: Ah!
I added it initially because I had some difficulty understanding how the `title` was used. if I placed the title inside the H2, it caused problems on the inner panels. It is not needed anymore, I will switch it back.

File: packages/edit-site/src/components/sidebar-navigation-screen/index.js [hunk 2]
```diff
@@ -40,10 +40,15 @@ export default function SidebarNavigationScreen( {
 					) : (
 						<div className="edit-site-sidebar-navigation-screen__icon-placeholder" />
 					) }
-
-					<div className="edit-site-sidebar-navigation-screen__title">
-						{ title }
-					</div>
+					{ ! parentTitle ? (
```
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar-navigation-screen/index.js]: Is there any reason to limit the H2 to only the top level?

File: packages/edit-site/src/components/sidebar-navigation-screen/index.js [hunk 3]
```diff
@@ -40,10 +40,15 @@ export default function SidebarNavigationScreen( {
...
+						<div className="edit-site-sidebar-navigation-screen__title">
```
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar-navigation-screen/index.js]: Should we just make this div an h2 to avoid the extra DOM element?

Comment (youknowriad) [reply packages/edit-site/src/components/sidebar-navigation-screen/index.js]: Also since it's exactly the same element tree. you could avoid the ternary by doing `TagName = parentTitle ? 'div' : 'h2';` and `<TagName....`

File: packages/edit-site/src/components/sidebar-navigation-screen/index.js (hunk 2 above)
Comment (carolinan) [reply packages/edit-site/src/components/sidebar-navigation-screen/index.js]: It is to avoid using the h2 as a wrapper for both the text and the add new button on the inner screen.
It is valid HTML, but when using Voiceover the heading text and button is announced twice.

Comment (youknowriad) [reply packages/edit-site/src/components/sidebar-navigation-screen/index.js]: Should we introduce a dedicated prop for the "new button", like an "actions" prop for instance?

Comment (carolinan) [reply packages/edit-site/src/components/sidebar-navigation-screen/index.js]: To place it outside of edit-site-sidebar-navigation-screen__title?

Comment (youknowriad) [reply packages/edit-site/src/components/sidebar-navigation-screen/index.js]: yes that's the idea

File: packages/edit-site/src/components/sidebar-navigation-screen/style.scss [hunk 4]
```diff
@@ -24,6 +24,15 @@
 	font-size: calc(1.56 * 13px);
 	font-weight: 500;
 	flex-grow: 1;
+	color: $white;
+	margin: 0;
+}
+
+.edit-site-sidebar-navigation-screen__title h2 {
```
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar-navigation-screen/style.scss]: This is also probably useless CSS now

File: packages/edit-site/src/components/sidebar-navigation-screen-templates/index.js [hunk 5]
```diff
@@ -119,21 +116,17 @@ export default function SidebarNavigationScreenTemplates( {
...
-							templateType={ postType }
-							toggleProps={ {
-								className:
-									'edit-site-sidebar-navigation-screen-templates__add-button',
-							} }
-						/>
-					) }
-				</HStack>
+			title={ config[ postType ].labels.title }
+			action={
```
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar-navigation-screen-templates/index.js]: For consistency, I'd probably rename this prop to `actions` as we've been using a similar prop in other components.

**Review by youknowriad — APPROVED** [review]
Really cool improvement to the component and a11 fix
